### PR TITLE
Update docs to specify supported version of kind

### PR DESCRIPTION
**Pull request by akhil-rane**
Currently, we support 0.5.1 version of kind. Our development doc does not mention
it. This PR updates the doc specifying the supported version of kind.

### CODE REVIEW

**Comment by gregsheremeta** [comment]
Can you also add a note about it in the "Deploying with Kubernetes In Docker (kind)" section? It's easy to miss in the bullet list, especially for "speed readers" like me :laughing:

File: docs/developing.md [hunk 1]
```diff
@@ -39,6 +39,7 @@
 ### External tools
 
 - [kustomize](https://github.com/kubernetes-sigs/kustomize#kustomize)
+- [kind](https://github.com/kubernetes-sigs/kind) (0.5.1)
```
Comment (staebler) [inline docs/developing.md:42]: I would rather see the version limitation spelled out explicitly in the `Deploying with Kubernetes In Docker` section. The kind tool is specific to that section.

Comment (staebler) [reply docs/developing.md:42]: In other words, I do not think that kind should be added to the `External Tools`.

Comment (akhil-rane) [reply docs/developing.md]: Done

**Pull request merged**